internal/cmd/gocdk/build.go

```diff
@@ -17,42 +17,54 @@ package main
 import (
 	"bytes"
 	"context"
-	"flag"
 	"io/ioutil"
 	"os/exec"
 	"path/filepath"
 	"strings"
 
+	"github.com/spf13/cobra"
 	"golang.org/x/xerrors"
 )
 
-func build(ctx context.Context, pctx *processContext, args []string) error {
-	f := newFlagSet(pctx, "build")
-	list := f.Bool("list", false, "display Docker images of this project")
-	ref := f.String("t", ":latest", "name and/or tag in the form `name[:tag] OR :tag`")
-	if err := f.Parse(args); xerrors.Is(err, flag.ErrHelp) {
-		return nil
-	} else if err != nil {
-		return usagef("gocdk build: %w", err)
-	}
-	if *list {
-		if err := listBuilds(ctx, pctx, f.Args()); err != nil {
-			return xerrors.Errorf("gocdk build: %w", err)
-		}
-		return nil
+const defaultDockerTag = ":latest"
+
+func registerBuildCmd(ctx context.Context, pctx *processContext, rootCmd *cobra.Command) {
+	var list bool
+	var ref string
+	buildCmd := &cobra.Command{
+		Use:   "build",
+		Short: "TODO Build a Docker image",
+		Long:  "TODO more about build",
+		Args:  cobra.ExactArgs(0),
+		RunE: func(_ *cobra.Command, _ []string) error {
+			if list {
+				if err := listBuilds(ctx, pctx); err != nil {
+					return xerrors.Errorf("gocdk build: %w", err)
+				}
+				return nil
+			}
+			return build(ctx, pctx, ref)
+		},
 	}
+	buildCmd.Flags().BoolVar(&list, "list", false, "display Docker images of this project")
+	buildCmd.Flags().StringVarP(&ref, "tag", "t", defaultDockerTag, "name and/or tag in the form `name[:tag] OR :tag`")
+	rootCmd.AddCommand(buildCmd)
+}
+
+// TODO(rvangent): Rename ref -> dockerTag for consistency?
```

Comment:
👍

Reply:
... It's complicated. `docker build`'s `-t` flag, which this flag is trying to keep consistency with, actually is specifying a [Docker image reference](https://godoc.org/github.com/docker/distribution/reference), not just the tag. I used to use the term "image name" to refer to this concept, but as @clausti has pointed out, that is imprecise. So throughout, I am trying to use "image reference" to mean the thing documented at the link above.

Reply:
I updated the TODO with a link to this comment.